bot/exts/holidays/holidayreact.py

```diff
@@ -0,0 +1,140 @@
+import logging
+import random
+import re
+from typing import NamedTuple
+
+import discord
+from discord.ext.commands import Cog
+
+from bot.bot import Bot
+from bot.constants import Month
+from bot.utils import resolve_current_month
+from bot.utils.decorators import in_month
+
+log = logging.getLogger(__name__)
+
+
+class Trigger(NamedTuple):
+    """Representation of regex trigger and corresponding reactions."""
+
+    regex: str
+    reaction: list[str]
+
+
+class Holiday(NamedTuple):
+    """Representation of reaction holidays."""
+
+    months: list[Month]
+    triggers: dict[str, Trigger]
+
+
+Valentines = Holiday([Month.FEBRUARY], {
+    "heart": Trigger(r"\blove|heart\b", ["\u2764\uFE0F"]),
+    }
+)
+Easter = Holiday([Month.APRIL], {
+    "bunny": Trigger(r"\beaster|bunny|rabbit\b", ["\U0001F430", "\U0001F407"]),
+    "egg": Trigger(r"\begg\b", ["\U0001F95A"]),
+    }
+)
+EarthDay = Holiday([Month.FEBRUARY], {
+    "earth": Trigger(r"\bearth|planet\b", ["\U0001F30E", "\U0001F30D", "\U0001F30F"]),
+    }
+)
+Pride = Holiday([Month.JUNE], {
+    "pride": Trigger(r"\bpride\b", ["\U0001F3F3\uFE0F\u200D\U0001F308"]),
+    }
+)
+Halloween = Holiday([Month.OCTOBER], {
+    "bat": Trigger(r"\bbat((wo)?m[ae]n|persons?|people|s)?\b", ["\U0001F987"]),
+    "danger": Trigger(r"\bdanger\b", ["\U00002620"]),
+    "doot": Trigger(r"\bdo{2,}t\b", ["\U0001F480"]),
+    "halloween": Trigger(r"\bhalloween\b", ["\U0001F383"]),
+    "jack-o-lantern": Trigger(r"\bjack-o-lantern\b", ["\U0001F383"]),
+    "pumpkin": Trigger(r"\bpumpkin\b", ["\U0001F383"]),
+    "skeleton": Trigger(r"\bskeleton\b", ["\U0001F480"]),
+    "spooky": Trigger(r"\bspo{2,}[k|p][i|y](er|est)?\b", ["\U0001F47B"]),
+    }
+)
+Hanukkah = Holiday([Month.NOVEMBER, Month.DECEMBER], {
+    "menorah": Trigger(r"\bc?hanukkah|menorah\b", ["\U0001F54E"]),
+    }
+)
+Christmas = Holiday([Month.DECEMBER], {
+    "christmas tree": Trigger(r"\b(christ|x)mas|tree\b", ["\U0001F384"]),
+    "reindeer": Trigger(r"\breindeer|caribou|buck|stag\b", ["\U0001F98C"]),
```

Comment:
I don't think this will work quite as intended with the word boundary, as e.g. the `\b` at the start is only part of `reindeer`, so while this prevents `areindeer` from matching, `acaribou` etc. would still match.

A fix for this would be to write `\b(reindeer|caribou|buck|stag)\b`, this also applies to a couple of the other regexes.